org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelCompletionParticipant.java

```diff
@@ -352,15 +372,29 @@ private void fillAttributeValuesWithCMAttributeDeclarations(CMElementDeclaration
 				item.setKind(CompletionItemKind.Value);
 				item.setFilterText(insertText);
 				item.setTextEdit(Either.forLeft(new TextEdit(fullRange, insertText)));
-				MarkupContent documentation = XMLGenerator.createMarkupContent(cmAttribute, value, cmElement, request);
-				item.setDocumentation(documentation);
+
+				if (request.isResolveDocumentationSupported()) {
+					DOMDocument document = request.getNode().getOwnerDocument();
+					JsonObject data = DataEntryField.createData(document.getDocumentURI(),
+							AttributeValueCompletionResolver.PARTICIPANT_ID);
+					try {

```

Comment:
I have 2 questions:

 * why can you not store the offset as int?
 * why you convert again the position to offset although you have it with request.getOffset();

In other words I would write:

data.addProperty(AttributeValueCompletionResolver.OFFSET_KEY, request.getOffset());

Reply:
Thanks for the pointers! This looks like a much better way to do this